src/graphql/models/User.js

```diff
@@ -158,6 +159,65 @@ const User = new GraphQLObjectType({
         'If not null, the user is blocked with the announcement in this string.',
       type: GraphQLString,
     },
+
+    badges: {
+      type: new GraphQLNonNull(
+        new GraphQLList(new GraphQLNonNull(UserAwardedBadge))
+      ),
+      description: 'Badges awarded to the user.',
+      resolve: (user) => user.badges || [],
+    },
+
+    majorBadgeBorderUrl: {
+      type: GraphQLString,
+      description: 'returns badge background image url',
+      resolve: async (user, args, { loaders }) => {
+        const displayItem = user.badges.find({ isDisplay: true });
+        if (displayItem == null) {
+          return null;
+        }
+        const badgeId = displayItem.id;
+        const badgeInfo = loaders.docLoader.load({
+          index: 'badges',
+          id: badgeId,
+        });
+        return badgeInfo.borderImage;
+      },
+    },
+
+    majorBadgeImageUrl: {
+      type: GraphQLString,
+      description: 'returns badge background image url',
+      resolve: async (user, args, { loaders }) => {
+        const displayItem = user.badges.find({ isDisplay: true });
+        if (displayItem == null) {
+          return null;
+        }
+        const badgeId = displayItem.id;
+        const badgeInfo = loaders.docLoader.load({
+          index: 'badges',
+          id: badgeId,
+        });
+        return badgeInfo.icon;
+      },
+    },
+
+    majorBadgeName: {
+      type: GraphQLString,
+      description: 'returns badge background image url',
+      resolve: async (user, args, { loaders }) => {
+        const displayItem = user.badges.find({ isDisplay: true });
+        if (displayItem == null) {
+          return null;
+        }
+        const badgeId = displayItem.id;
+        const badgeInfo = loaders.docLoader.load({
+          index: 'badges',
+          id: badgeId,
+        });
+        return badgeInfo.name;
+      },
+    },
```

Comment:
It is recommended that we add a unit test (and relevant fixtures) under https://github.com/cofacts/rumors-api/blob/master/src/graphql/queries/__tests__/GetUser.js to ensure that these newly added fields can be loaded as expected.